llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/IR/LinalgExtOps.cpp

```diff
@@ -1180,6 +1180,82 @@ Operation *ReverseOp::getTiledImplementation(OpBuilder &builder,
   return tiledRevOp;
 }
 
+//===----------------------------------------------------------------------===//
+// TopkOp
+//===----------------------------------------------------------------------===//
+
+LogicalResult TopkOp::verify() {
+  Operation *op = getOperation();
+  if (getNumInputs() != 2) {
+    return op->emitOpError("expected two input operands");
+  }
+  if (getNumOutputs() != 2) {
+    return op->emitOpError("expected two output operands");
+  }
+  if (!inputValues().getType().isa<ShapedType>() ||
+      !inputIndices().getType().isa<ShapedType>()) {
+    return op->emitOpError("expected input element types to be shaped");
+  }
+  if (!outputValues().getType().isa<ShapedType>() ||
+      !outputIndices().getType().isa<ShapedType>()) {
+    return op->emitOpError("expected output element types to be shaped");
+  }
+  // Ensure input/output values types match
+  ShapedType inputValuesType = inputValues().getType().cast<ShapedType>();
+  ShapedType outputValuesType = outputValues().getType().cast<ShapedType>();
```

Comment:
style nit: Use `auto` because `cast` already spells the type.

https://llvm.org/docs/CodingStandards.html#use-auto-type-deduction-to-make-code-more-readable